src/ert/shared/models/base_run_model.py

```diff
@@ -138,12 +138,16 @@ def _create_mask_from_failed_realizations(self) -> List[bool]:
         Creates a list of bools representing the failed realizations,
         i.e., a realization that has failed is assigned a True value.
         """
-        return [
-            initial and not completed
-            for initial, completed in zip(
-                self._initial_realizations_mask, self._completed_realizations_mask
-            )
-        ]
+        if len(self._completed_realizations_mask) == 0:
```

Comment:
Can we change this to `if self._completed_realizations_mask is None` instead, and then have this attribute `None` from object initialization and also set to `None` instead of `[]` in the case of an `ErtRunError` (which I speculate is happening here?)

Reply:
If possible (meaning that the ensemble length is always known) we can also change it to always have the correct length (but it is maybe needed to allow `None` at init?)

Reply:
> Can we change this to `if self._completed_realizations_mask is None` instead, and then have this attribute `None` from object initialization and also set to `None` instead of `[]` in the case of an `ErtRunError` (which I speculate is happening here?)

We would then also have to change the input argument to BaseRunModel to Optional[List[bool]] = None right?
Not sure we gain that much from using None instead of [] here.

Reply:
Right. A zero-length ensemble is always an error in Ert (?) so `[]` is "free" to be used equivalent to `None`. It might be cleaner still to use `None` as `[]` could be thought of as ok if one believes zero-length ensembles is a thing.

Reply:
So, I tried running poly.ert with 0 realisations and when trying to run an ensemble experiment get this crash:

ValueError: Wrong range syntax 0--1

😬

Reply:
👍🏻

Reply:
Using None would mean that we also have to change `_count_successful_realizations` I think, since it's using `.count`.
Think we'll just leave as is and use [].

Reply:
that is ok!

Reply:
(but pylint frowns upon `if len(list) == 0`, do `if list` instead.)

Reply:
Pylint does not like bare len(list), explicit checks like len(list)==0 is fine according to pep8.
This was considered a bug in pylint:
https://github.com/PyCQA/pylint/pull/2815

Reply:
ah, but pep8 still says that `if not list` is good in this situation?

Reply:
I'm fine with either as long as the bug is fixed.
I've pushed a version that uses `if list`.